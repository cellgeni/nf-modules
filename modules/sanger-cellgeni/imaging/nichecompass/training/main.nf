process NICHECOMPASS_TRAINING {
    tag "${meta.id}"
    label 'process_gpu'

    container "quay.io/cellgeni/nichecompass:0.3.0"

    input:
    tuple val(meta), path(h5ad, stageAs: "inputs/*")

    output:
    tuple val(meta), path("${prefix}_*"), emit: nichecompass_dir  // TODO is parsing path with glob fine?
    tuple val(meta), path("${prefix}_*/timestamp.txt"), emit: timestamp
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    nichecompass_train_sample_integration.py \\
        --batches ${h5ad} \\
        --prefix "${prefix}" \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        nichecompass: \$(pip show nichecompass | grep Version | sed -e "s/Version: //g")
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    def timestamp = new Date().format("yyyyMMdd_HHmmss")
    """
    mkdir -p "${prefix}_${timestamp}"
    mkdir -p "${prefix}_${timestamp}/artifacts/sample_integration/nichecompass/figures"
    mkdir -p "${prefix}_${timestamp}/artifacts/sample_integration/nichecompass/models"
    mkdir -p "${prefix}_${timestamp}/data"
    touch "${prefix}_${timestamp}/artifacts/sample_integration/nichecompass/models/model.h5ad"
    touch "${prefix}_${timestamp}/train.log"
    touch "${prefix}_${timestamp}/timestamp.txt"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        nichecompass: \$(pip show nichecompass | grep Version | sed -e "s/Version: //g")
    END_VERSIONS
    """
}
