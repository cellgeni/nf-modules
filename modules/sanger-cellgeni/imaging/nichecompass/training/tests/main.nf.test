nextflow_process {

    name "Test Process NICHECOMPASS_TRAINING"
    script "../main.nf"
    process "NICHECOMPASS_TRAINING"

    tag "modules"
    tag "modules_sanger-cellgeni"
    tag "nichecompass"
    tag "nichecompass/training"

    test("mouse - 3") {

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '*.h5ad', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("mouse - 3 - stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + '*.h5ad', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
