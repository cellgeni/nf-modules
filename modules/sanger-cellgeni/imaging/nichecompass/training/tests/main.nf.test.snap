{
    "mouse - 3 - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "nichecompass_outdir": [
                    
                ],
                "timestamp": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-08-21T14:51:34.028585"
    },
    "mouse - 3": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "nichecompass_outdir": [
                    
                ],
                "timestamp": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-08-21T14:51:28.885423"
    }
}